crates/net/discv4/src/lib.rs

```diff
@@ -432,6 +432,11 @@ pub struct Discv4Service {
     queued_pings: VecDeque<(NodeRecord, PingReason)>,
     /// Currently active pings to specific nodes.
     pending_pings: HashMap<PeerId, PingRequest>,
+    /// Currently active endpoint proof verification lookups to specific nodes.
+    ///
+    /// Entries here means we've proven the peer's endpoint but haven't completed our end of the
+    /// endpoint proof
+    pending_lookup: HashMap<PeerId, LookupContext>,
```

Comment:
we need to timestamp these and evict these on an interval basis, I think we can reuse the existing interval 

https://github.com/paradigmxyz/reth/blob/4dceabf06b9ef26422264ad2026f6e23fda38fbb/crates/net/discv4/src/lib.rs#L449-L450

because otherwise entries might not get deleted